types/id.cc

```diff
@@ -157,6 +157,11 @@ parse(const char * value, size_t len, Type type)
                 unsigned long long val = 0;
                 for (unsigned i = start;  i != start + len;  ++i) {
                     int c = p[i];
+                    if (c >= 65 && c <= 70) {
```

Comment:
For more clarity and readability, I would write:

```cpp
if (c >= 'A' && c <= 'F') {
    ...
}
```

For maximum portability, I would check against every character since the standard does not guarantee that 'A' + 1 == 'B' but that's minor, I don't think our goal is to be non-ascii compatible anyway :)